File: src/ios/LukuIDSDK/Sources/LukuIDSDK/DebugLog.swift
// SPDX-License-Identifier: Apache-2.0
import Foundation

@inline(__always)
func debugLog(_ options: LukuIDClientOptions, _ message: String, context: [String: Any] = [:]) {
    guard options.debugLogging else {
        return
    }

    if context.isEmpty {
        print("[LukuIDSDK] \(message)")
        return
    }

    print("[LukuIDSDK] \(message) \(context)")
}

File: src/android/lukuid-sdk/src/main/java/com/lukuid/sdk/internal/DebugLog.kt
// SPDX-License-Identifier: Apache-2.0
package com.lukuid.sdk.internal

import android.util.Log
import com.lukuid.sdk.LukuSdkOptions

private const val TAG = "LukuSdk"

internal fun debugLog(options: LukuSdkOptions, message: String, context: Map<String, Any?> = emptyMap()) {
    if (!options.debugLogging) {
        return
    }

    if (context.isEmpty()) {
        Log.d(TAG, message)
        return
    }

    Log.d(TAG, "$message $context")
}
